ui/components/ui/button/buttons.scss

```diff
@@ -134,8 +116,7 @@
 
   &--disabled,
   &[disabled] {
-    opacity: 1;
-    color: var(--color-text-muted);
+    opacity: var(--opacity-disabled);
```

Comment:
Button default adheres to design system standards and works as expected when comparing against develop github storybook


https://github.com/MetaMask/metamask-extension/assets/8112138/e952318e-2961-42ca-8961-77cb0f865b00